src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -255,6 +255,49 @@ public void start(final Map<String, String> props) {
     return null;
   }
 
+  private Optional<SourceRecord> createSourceRecord(
+      final Map<String, Object> partition,
+      final SchemaAndValueProducer keySchemaAndValueProducer,
+      final SchemaAndValueProducer valueSchemaAndValueProducer,
+      final Map<String, String> sourceOffset,
+      final String topicName,
+      final BsonDocument valueDoc,
+      final BsonDocument keyDocument) {
+
+    try {
+      SchemaAndValue keySchemaAndValue = keySchemaAndValueProducer.get(keyDocument);
+      SchemaAndValue valueSchemaAndValue = valueSchemaAndValueProducer.get(valueDoc);
+      return Optional.of(
+          new SourceRecord(
+              partition,
+              sourceOffset,
+              topicName,
+              keySchemaAndValue.schema(),
+              keySchemaAndValue.value(),
+              valueSchemaAndValue.schema(),
+              valueSchemaAndValue.value()));
+    } catch (Exception e) {
+      if (sourceConfig.tolerateErrors()) {
+        if (sourceConfig.getBoolean(ERRORS_LOG_ENABLE_CONFIG)) {
```

Comment:
Why aren't log levels controlled in the usual way that one does for SLF4J (by configuring log levels in your SLF4J provider config, e.g. logback.xml)

Reply:
This is a Kafka connect configuration that we are borrowing. It allows errors to be ignored and logged and not passed to a DLQ.